### PR TITLE
Limit results generated by SELECT queries in MSQ

### PR DESCRIPTION
**Pull request by LakshSingla**
### Description

MSQ queries return the results in the query reports. Querying large data sources can generate a lot of rows, all of which will get materialized in the query reports, causing them to grow very large. This PR limits the results of the SELECT queries that we put in the reports.

Note: Only way to fetch the complete result set after this change would be to run the query using the native engine, however, we might add a way to execute long-running queries that post their complete results in a different. Therefore the rows in the reports would serve as a preview to the actual result set.

#### Release note
SELECT queries executed using MSQ generate only a subset of the results in the query reports. 

<hr>

##### Key changed/added classes in this PR
 * `ControllerImpl`

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [x] a release note entry in the PR description.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by adarshsanjeev — APPROVED** [review]
LGTM! Thanks for including the patch notes too.

File: extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java [hunk 1]
```diff
@@ -1393,67 +1393,69 @@ private Yielder<Object[]> getFinalResultsYielder(
...
+                             }
+                           }
+                           retVal.add(row);
+                           cursor.advance();
+                         }
+
+                         return Sequences.simple(retVal);
+                       }
+                   )
+                   .limit(Limits.MAX_SELECT_RESULT_ROWS)
```
Comment (cryptoe) [inline extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java:1457]: For the end user, we need to set a flag in the report mentioning the results are truncated.

**Comment by cryptoe** [comment]
Please add the truncated flag in the read me. 
LGTM to me otherwise.

**Review by cryptoe — APPROVED** [review]

**Comment by cryptoe** [comment]
cc @vogievetsky We now have a new field inside the task report : `multiStageQuery.payload.results.resultsTruncated` which denotes if the results are truncated.

**Pull request merged**